python/lsst/obs/comCam/ingest.py

```diff
@@ -18,14 +18,30 @@ def __init__(self, config, *args, **kwargs):
         super(ParseTask, self).__init__(config, *args, **kwargs)
 
     def getInfo(self, filename):
-        # Grab the basename
+        """ Get the basename and other data which is only available from the filename/path.
+        
+        This seems fragile, but this is how the teststand data will *always* be written out, 
+        as the software has been "frozen" as they are now in production mode.
+
+        Parameters
+        ----------
+        filename : `str`
+            The filename
+
+        Returns
+        -------
+        phuInfo : `dict`
+            Dictionary containing the header keys defined in the ingest config from the primary HDU
+        infoList : `list`
+            A list of dictionaries containing the phuInfo(s)
```

Comment:
In the case of single extension files, it's a dict, and a [dict], otherwise the second object is a list of dictionaries containing header data. I agree it's not very nice, but again, it's inheriting this structure from the base class. All this should be cleaned up in the Great Obs Package Redesign, but for now I think this is how it has to be 😔